### DISCUSSION TITLE
Is there a native way to use the same `require.resolve` algorithm to resolve closest packages from nested directories?

### DISCUSSION THREAD

**Discussion by edumudu** [nodejs/help#4953] [Q&A]
Basically, I want to resolve the `package.json` path from any directory inside the project. I know that I can make a custom function to get this, but I would like to know if there is a native way of use this same algorithm to find files.

The node exposes some API to do this with files?

**Comment by aduh95** [comment]
There's `createRequire` that may fit your use-case: https://nodejs.org/api/module.html#modulecreaterequirefilename
